File: server/src/main/java/io/kragelv/library/dto/book/BookDto.java
package io.kragelv.library.dto.book;

import java.util.List;

import io.kragelv.library.dto.genre.GenreDto;

public record BookDto(
        String id,
        String title,
        AuthorDto author,
        Integer publishedYear,
        Integer availableCopies,
        List<GenreDto> genres) {

    public record AuthorDto(
            String id,
            String firstName,
            String middleName,
            String lastName) {
    }
}

File: client/src/dtos/book/BookDto.ts
import { AuthorDto } from '../author/AuthorDto';
import { GenreDto } from '../genre/GenreDto';

export interface BookDto {
  id: string;
  title: string;
  author: AuthorDto;
  publishedYear: number;
  availableCopies: number;
  genres: GenreDto[];
}
